### PR TITLE
PUD-752: Use only First and Last name everywhere when showing prisoner name

### PR DESCRIPTION
**Pull request by philsegal**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/dossier/TableOfContentsGeneratorTest.kt [hunk 1]
```diff
@@ -67,7 +67,7 @@ class TableOfContentsGeneratorTest {
           { it.variableAsString("recallLengthAndSentenceHeading") },
           equalTo(expectedText)
         ),
-        has("fullName", { it.variableAsString("fullName") }, equalTo("Bertie Basset Badger")),
+        has("fullName", { it.variableAsString("fullName") }, equalTo("Bertie Badger")),
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/dossier/TableOfContentsGeneratorTest.kt:70]: This isn't really "fullName" anymore - perhaps "personName" would be better.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationContextFactoryTest.kt [hunk 2]
```diff
@@ -102,7 +102,6 @@ class RecallNotificationContextFactoryTest {
     val lastReleasePrisonName = PrisonName("Last Prison Name")
     val prisoner = Prisoner(
       firstName = "Bobbie",
-      middleNames = "Bertrude",
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationContextFactoryTest.kt:105]: Not sure I'd delete this line; it's still valid input - both ways of course.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationContextTest.kt [hunk 3]
```diff
@@ -78,12 +78,11 @@ class RecallNotificationContextTest {
   )
 
   private val prisoner = Prisoner(
+    croNumber = prisonerCroNumber,
     firstName = "Bertie",
-    middleNames = "Basset",
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationContextTest.kt:82]: Ditto - Not sure I'd delete this line; it's still valid input - both ways of course.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/documents/GotenbergComponentTestBase.kt [hunk 4]
```diff
@@ -124,11 +124,11 @@ abstract class GotenbergComponentTestBase : ComponentTestBase(useRealGotenbergSe
       listOf(
         Prisoner(
           prisonerNumber = nomsNumber.value,
+          croNumber = "CRO Num/456",
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/documents/GotenbergComponentTestBase.kt:127]: I'm just wondering why you re-ordered all these croNumber and pncNumber settings?

**Review by AndyNewton — COMMENTED** [review]
Few minor QQs.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/documents/GotenbergComponentTestBase.kt (hunk 4 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/documents/GotenbergComponentTestBase.kt:127]: I initially did a refactor on Prisoner to remove middleName. I'm guessing the IntelliJ function then ordered them according to the constructor.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationContextFactoryTest.kt (hunk 2 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationContextFactoryTest.kt:105]: Can see that, but in unit tests we generally only populate properties that are relevant to the test.

**Pull request merged**